lib/rucio/daemons/reaper/reaper.py

```diff
@@ -711,12 +711,10 @@ def run(
         raise DatabaseException('Database was not updated, daemon won\'t start')
 
     logging.log(logging.INFO, 'main: starting processes')
-    rses_to_process = get_rses_to_process(rses, include_rses, exclude_rses, vos)
-    if not rses_to_process:
-        logging.log(logging.ERROR, 'Reaper: No RSEs found. Exiting.')
-        return
 
-    logging.log(logging.INFO, 'Reaper: This instance will work on RSEs: %s', ', '.join([rse['rse'] for rse in rses_to_process]))
+    while not get_rses_to_process(rses, include_rses, exclude_rses, vos):
+        logging.log(logging.WARNING, 'Reaper: No RSEs found. Sleeping for %s seconds and retrying.', sleep_time)
+        time.sleep(sleep_time)
```

Comment:
Repeating my first comment, what is the current purpose of this block?

Let’s say that you take it out entirely. Inside `run_once()`, `rses_to_process` will be calculated, and if it is empty, if will write a log entry and sleep (note: specifying 30 seconds is incorrect and the level should have been a warning and not an error). If not empty, it will even write the list of RSEs it will work on and get to work. In short, the Reaper will likely work just fine.

So, what are we losing?

Reply:
Now I understand better what you mean, addressed it in https://github.com/rucio/rucio/pull/7472/commits/58940f80a3a14b8df136916afec4bc361cdee07f.

Now threads are spawned even if no RSE are available.